lisa/sut_orchestrator/azure/features.py

```diff
@@ -203,41 +195,135 @@ def _initialize(self, *args: Any, **kwargs: Any) -> None:
+            capability is None or capability.disk_type is None
+        ):
+            result.add_reason("capability doesn't have disk_type.")
+
+        if self.disk_type:
+            has_meet_disk_type = False
+            if isinstance(self.disk_type, schema.DiskType):
+                req_disk_types = search_space.SetSpace[schema.DiskType](
+                    items=[self.disk_type]
+                )
+            else:
+                req_disk_types = self.disk_type
+            for req_disk_type in req_disk_types:
+                if isinstance(capability.disk_type, schema.DiskType):
+                    if req_disk_type == capability.disk_type:
+                        has_meet_disk_type = True
+                        break
+                else:
+                    assert isinstance(capability.disk_type, search_space.SetSpace)
+                    if req_disk_type in capability.disk_type:
+                        has_meet_disk_type = True
+                        break
+            if not has_meet_disk_type:
+                result.add_reason(
+                    f"no disk type supported in capability. "
+                    f"requirement: {self.disk_type}, "
+                    f"capability: {capability.disk_type}"
+                )
 
-class DiskPremiumLRS(AzureFeatureMixin, features.DiskPremiumLRS):
-    def _initialize(self, *args: Any, **kwargs: Any) -> None:
-        super()._initialize(*args, **kwargs)
-        self._initialize_information(self._node)
+        return result
 
-    @staticmethod
-    def get_disk_id() -> str:
-        return "Premium_LRS"
+    def generate_min_capability(self, capability: Any) -> Any:
+        assert isinstance(
+            capability, AzureDiskOptionSettings
+        ), f"actual: {type(capability)}"
+        assert (
+            capability.disk_type
+        ), "capability should have at least one disk type, but it's None"
+        cap_disk_type = capability.disk_type
+        if isinstance(cap_disk_type, search_space.SetSpace):
+            assert (
+                len(cap_disk_type) > 0
+            ), "capability should have at least one disk type, but it's empty"
+        elif isinstance(cap_disk_type, schema.DiskType):
+            cap_disk_type = search_space.SetSpace[schema.DiskType](
+                is_allow_set=True, items=[cap_disk_type]
+            )
+        else:
+            raise LisaException(
+                f"unknown disk type on capability, type: {cap_disk_type}"
+            )
 
+        # if there is no requirement, copy capability to get min one.
+        if self.disk_type is None:
+            req_disk_type = capability.disk_type
+        else:
+            req_disk_type = self.disk_type
+
+        # find the min disk type from the order by cost.
+        min_disk_type: Optional[schema.DiskType] = None
+        for expected_disk_type in _ordered_disk_types:
+            if (
+                expected_disk_type in req_disk_type
```

Comment:
Is this loop required? Wouldn't `req_disk_type` be set to `StandardHDDLRS` if tests don't specify anything and all azure VM's support this?